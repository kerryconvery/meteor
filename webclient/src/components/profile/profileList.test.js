import React from 'react';
import { shallow } from 'enzyme';
import ProfileList from './profileList';

test('renders a list of profiles', () => {
  const media = [
    {
      name: 'profile1',
    },
    {
      name: 'profile2',
    },
  ];
  const wrapper = shallow(<ProfileList items={media} />);
  expect(wrapper).toMatchSnapshot();
});

