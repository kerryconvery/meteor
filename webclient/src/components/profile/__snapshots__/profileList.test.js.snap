// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of profiles 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileList
    items={
      Array [
        Object {
          "name": "profile1",
        },
        Object {
          "name": "profile2",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          <NavLink
            activeClassName="active"
            aria-current="page"
            exact={true}
            to="/media?profile=profile1"
          >
            profile1
          </NavLink>
        </li>,
        <li>
          <NavLink
            activeClassName="active"
            aria-current="page"
            exact={true}
            to="/media?profile=profile2"
          >
            profile2
          </NavLink>
        </li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "profile1",
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            exact={true}
            to="/media?profile=profile1"
          >
            profile1
          </NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "profile1",
            "exact": true,
            "to": "/media?profile=profile1",
          },
          "ref": null,
          "rendered": "profile1",
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "profile2",
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            exact={true}
            to="/media?profile=profile2"
          >
            profile2
          </NavLink>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "profile2",
            "exact": true,
            "to": "/media?profile=profile2",
          },
          "ref": null,
          "rendered": "profile2",
          "type": [Function],
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              exact={true}
              to="/media?profile=profile1"
            >
              profile1
            </NavLink>
          </li>,
          <li>
            <NavLink
              activeClassName="active"
              aria-current="page"
              exact={true}
              to="/media?profile=profile2"
            >
              profile2
            </NavLink>
          </li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "profile1",
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              exact={true}
              to="/media?profile=profile1"
            >
              profile1
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "profile1",
              "exact": true,
              "to": "/media?profile=profile1",
            },
            "ref": null,
            "rendered": "profile1",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "profile2",
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              exact={true}
              to="/media?profile=profile2"
            >
              profile2
            </NavLink>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "profile2",
              "exact": true,
              "to": "/media?profile=profile2",
            },
            "ref": null,
            "rendered": "profile2",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
